chainer/training/extensions/log_report.py

```diff
@@ -90,21 +97,25 @@ def __call__(self, trainer):
             if self._postprocess is not None:
                 self._postprocess(stats_cpu)
 
-            self._log.append(stats_cpu)
-
-            # write to the log file
-            if self._log_name is not None:
-                log_name = self._log_name.format(**stats_cpu)
-                fd, path = tempfile.mkstemp(prefix=log_name, dir=trainer.out)
-                with os.fdopen(fd, 'w') as f:
-                    json.dump(self._log, f, indent=4)
+            if self._trigger(trainer):
+                self._log.append(stats_cpu)
 
-                new_path = os.path.join(trainer.out, log_name)
-                shutil.move(path, new_path)
+            if self._write_trigger(trainer):
+                self._write(trainer)
```

Comment:
It seems the argument is inconsistent with function declaration of `_write`. The error is not raised in the unit test because `_write` method is substituted with a mock object.